test-basic/plan-search.js

```diff
@@ -166,8 +178,76 @@ describe('search', function() {
         should(output[1]['relevance.doc'].value.musician.instrument[0]).equal('trumpet');
         should(output[0]['relevance.score'].value).greaterThanOrEqual(output[1]['relevance.score'].value);
         done();
-      }).catch(done);
+      }).catch(error => done(error));
     });
 // console.log(JSON.stringify(output, null, 2));
   });
+
+  describe('tests for new scoring methods -  bm25, random and zero.', function() {
+    before(function (done) {
+      if(serverConfiguration.serverVersion < 12) {
+        this.skip();
+      }
+      done();
+    });
+
+    it('should search documents with bm25', function(done) {
+      execPlan(
+          p.fromSearch(p.cts.jsonPropertyValueQuery('instrument', 'trumpet'),
+              ['score', 'quality'], null, { scoreMethod: 'bm25', bm25LengthWeight: 0.25 })
+      ).then(function(response) {
+        assert(response.columns != null)
+        assert(response.rows != null)
+        done();
+      }).catch(error => done(error));
+    });
+
+    it('should throw error with invalid bm25LengthWeight', function(done) {
+      execPlan(
+          p.fromSearch(p.cts.jsonPropertyValueQuery('instrument', 'trumpet'),
+              ['score', 'quality'], null, { scoreMethod: 'bm25', bm25LengthWeight: 99 })
+      ).catch(error => {
+        try{
+          assert(error.body.errorResponse.message.toString().includes('Invalid option "bm25-length-weight=99'));
+          done();
+        } catch(error){
+          done(error);
+        }
+      });
+    });
+
+    it('should throw error with string values for bm25LengthWeight', function(done) {
+      try {
+        execPlan(
+            p.fromSearch(p.cts.jsonPropertyValueQuery('instrument', 'trumpet'),
+                ['score', 'quality'], null, { scoreMethod: 'bm25', bm25LengthWeight: 'abc' })
+        );
+      } catch(error){
+        assert(error.message.toString().includes('bm25LengthWeight must be a number'));
+        done();
+      }
+    });
+
+    it('should search documents with zero scoring method', function(done) {
+      execPlan(
+          p.fromSearch(p.cts.jsonPropertyValueQuery('instrument', 'trumpet'),
+              ['score', 'quality'], null, { scoreMethod: 'zero'})
+      ).then(function(response) {
+        assert(response.columns != null)
```

Comment:
This test is worth adding an assertion to verify that each row has a column named "score" with value of zero. That also helps ensure that `scoreMethod` is working regardless of the value.